desktop/ui/src/main/java/org/datacleaner/user/MutableServerInformationCatalog.java

```diff
@@ -0,0 +1,135 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.user;
+
+import org.datacleaner.configuration.DomConfigurationWriter;
+import org.datacleaner.configuration.ServerInformation;
+import org.datacleaner.configuration.ServerInformationCatalog;
+import org.datacleaner.util.StringUtils;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Mutable/modifyable implementation of the {@link ServerInformationCatalog} interface. Used to
+ * allow the user to change the catalog of ServerInformation at runtime. This server information
+ * catalog wraps an immutable instance, which typically represents what is
+ * configured in datacleaner's xml file.
+ */
+
+public class MutableServerInformationCatalog implements ServerInformationCatalog {
+    private final DomConfigurationWriter _configurationWriter;
+    private final List<ServerInformationChangeListener> _listeners = new LinkedList<>();
+
+    private final UserPreferences _userPreferences;
```

Comment:
The below comment means that this class shouldn't have ```UserPreferences``` at all. It should be able to just work with the configuration writer.